weave

```diff
@@ -562,13 +601,8 @@ populate_dns() {
         return 0
     fi
     for CONTAINER in $(docker ps -q --no-trunc); do
-        MORE_ARGS=$(docker inspect --format='--data-urlencode fqdn={{.Config.Hostname}}.{{.Config.Domainname}}.' $CONTAINER 2>/dev/null) && true
         if CONTAINER_IPS=$(with_container_netns $CONTAINER container_weave_addrs 2>&1 | sed -n -e 's/inet \([^/]*\)\/\(.*\)/\1/p') ; then
-            for IP in $CONTAINER_IPS; do
-                # NB: CONTAINER_IP is the IP of the weavedns
-                # container; it is set by wait_for_status.
-                http_call_ip $CONTAINER_IP $DNS_HTTP_PORT PUT /name/$CONTAINER/$IP $MORE_ARGS
-            done
+            tell_dns PUT $CONTAINER $CONTAINER_IPS
```

Comment:
Not convinced. I think the convenience of calling `tell_dns` wins out, as that is what is done everywhere else. So one extra call to `docker` to check if WeaveDNS is running, plus one more to get the ID, makes three per container instead of just the call to get the domain name which is present in both. So, we triple the number of `docker` calls

How many containers do you think need to be running per host before this will be an issue? I think that with small numbers (how many containers per host are people using, on average?) of containers it should not be noticeable, but it would be useful to know some exact numbers, probably.